scripts/c-btmInference.py

```diff
@@ -1,186 +1,226 @@
+    collected into a list and returned.
 
-        generated_tokens.append(next_token.item())
-        input_ids = torch.cat((input_ids, next_token), dim=-1)
+  Note: The models and tokenizers in the input lists should correspond to each other.
+    The length of indices, models, and tokenizers should be the same.
+  """
+  all_models_next_token_probs = []
 
-    generated_text = tokenizer.decode(generated_tokens)
-    return generated_text, chosen_token_probability
+  for index in range(len(indices)):
 
+      input_ids = tokenizers[index].encode(query, return_tensors="pt")
+      output = models[index](input_ids)
+      next_token_probs = output.logits[:, -1, :].softmax(dim=-1)
+      all_models_next_token_probs.append(next_token_probs)
 
+  return all_models_next_token_probs
 
 
-def findEnsembleWeights(embedder, prompt, T, clusterCenters):
-  """
-  takes in cluster centers and embedder using in clustering, and prompt, and T parameter
-  calculates weighted logit between context and cluster center
-  embedder: the embedder that was used to train the clustering model
-  prompt: the string prompt from input_file
-  T: temperature parameter for softmax
-  clusterCenters: list of cluster centers
+
+def findEnsembleWeights(embedder, query, clusterCenters, T):
   """
-  ensembleWeights = torch.tensor([])#[]
+  Finds ensemble weights based on distance of query and cluster centers.
+
+  Parameters:
+  - embedder: the embedder that was used to train the clustering model
+  - query: the string prompt from input_file
+  - clusterCenters: list of cluster centers
+  - T: temperature parameter for softmax
+
+  Returns:
+  list the distance between the queries and each of the cluster centers.
+  """  
+  ensembleWeights = []
+  embedded_query = embedder(query)) # make sure embedder returns tensor
   pdist = torch.nn.PairwiseDistance(p=2)
-  embeddedInput = embedder.encode(prompt)
-  
-  for domain, clusterCenter in enumerate(clusterCenters): # assuming modelList and clusterCenters have matching indices
-    ensembleWeight = 0
-
-    for token in embeddedInput[0]:
 
-      token = torch.tensor(token)
-      ensembleWeight += torch.exp(-1 * torch.pow(pdist(token, clusterCenter),2) / T)
-    # Check if ensembleWeights is empty, if so, initialize it
-    if ensembleWeights is None:
-        ensembleWeights = ensembleWeight
-    else:
-        # Check if ensembleWeights and ensembleWeight have compatible shapes
-        if ensembleWeights.ndim == 0:
-            # If ensembleWeights is a scalar, convert it to a 1-dimensional tensor
-            ensembleWeights = ensembleWeights.unsqueeze(0)
-        if ensembleWeight.ndim == 0:
-            # If ensembleWeight is a scalar, convert it to a 1-dimensional tensor
-            ensembleWeight = ensembleWeight.unsqueeze(0)
+  for cluster in clusterCenters:
+    ensembleWeights.append(torch.exp(-1 * torch.pow(pdist(embedded_query, cluster),2) / T).mean())

```

Comment:
need to take the average (or another aggregate statistic) as otherwise a tensor is returned for each token in the embedded query that presents the distance between itself and the cluster center